bigbluebutton-html5/imports/ui/components/presentation/container.jsx

```diff
@@ -1,29 +1,116 @@
 import PropTypes from 'prop-types';
-import { withTracker } from 'meteor/react-meteor-data';
 import { notify } from '/imports/ui/services/notification';
-import PresentationService from './service';
-import { Slides } from '/imports/api/slides';
 import Presentation from '/imports/ui/components/presentation/component';
 import PresentationToolbarService from './presentation-toolbar/service';
 import { UsersContext } from '../components-data/users-context/context';
 import Auth from '/imports/ui/services/auth';
-import Meetings from '/imports/api/meetings';
 import getFromUserSettings from '/imports/ui/services/users-settings';
 import {
   layoutSelect,
   layoutSelectInput,
   layoutSelectOutput,
   layoutDispatch,
 } from '../layout/context';
-import lockContextContainer from '/imports/ui/components/lock-viewers/context/container';
 import WhiteboardService from '/imports/ui/components/whiteboard/service';
 import { DEVICE_TYPE } from '../layout/enums';
 import MediaService from '../media/service';
+import { useSubscription } from '@apollo/client';
+import {
+  CURRENT_PRESENTATION_PAGE_SUBSCRIPTION,
+  CURRENT_PAGE_WRITERS_SUBSCRIPTION,
+} from '/imports/ui/components/whiteboard/queries';
+import POLL_SUBSCRIPTION from '/imports/ui/core/graphql/queries/pollSubscription';
+import useMeeting from '/imports/ui/core/hooks/useMeeting';
+
+const APP_CONFIG = Meteor.settings.public.app;
+const PRELOAD_NEXT_SLIDE = APP_CONFIG.preloadNextSlides;
+const fetchedpresentation = {};
+
+const PresentationContainer = (props) => {
+  const { data: presentationPageData } = useSubscription(CURRENT_PRESENTATION_PAGE_SUBSCRIPTION);
+  const { pres_page_curr: presentationPageArray } = (presentationPageData || {});
+  const currentPresentationPage = presentationPageArray && presentationPageArray[0];
+  const slideImageUrls = currentPresentationPage && JSON.parse(currentPresentationPage.urls);
+
+  const { data: whiteboardWritersData } = useSubscription(CURRENT_PAGE_WRITERS_SUBSCRIPTION);
+  const whiteboardWriters = whiteboardWritersData?.pres_page_writers || [];
+
+  const meeting = useMeeting((m) => ({
+    lockSettings: m?.lockSettings,
+  }));
+
+  const isViewersCursorLocked = meeting?.lockSettings?.hideViewersCursor || true;
+  const isViewersAnnotationsLocked = meeting?.lockSettings?.hideViewersAnnotation || true;
+
+  const multiUserData = {
+    active: whiteboardWriters?.length > 0,
+    size: whiteboardWriters?.length || 0,
+    hasAccess: whiteboardWriters?.some((writer) => writer.userId === Auth.userID),
+  };
+
+  const { data: pollData } = useSubscription(POLL_SUBSCRIPTION);
+  const poll = pollData?.poll[0] || {};
+
+  const currentSlide = currentPresentationPage ? {
+    content: currentPresentationPage.content,
+    current: currentPresentationPage.isCurrentPage,
+    height: currentPresentationPage.height,
+    width: currentPresentationPage.width,
+    id: currentPresentationPage.pageId,
+    imageUri: slideImageUrls.svg,
```

Comment:
I believe we switch to PNG in certain scenarios (when vector graphics too complex/large?) cc @gustavotrott may confirm?